ui/guest/src/communicator.ts

```diff
@@ -30,12 +30,12 @@ const broadcastChannel = useBroadcastChannel
 export const message$ = fromEvent(useBroadcastChannel ? broadcastChannel : window, 'message').pipe(share());
 
 export const post = useBroadcastChannel
-  ? (type, payload) => broadcastChannel.postMessage((typeof type === 'object') ? type : { type, payload })
-  : (type, payload) => window.parent.postMessage((typeof type === 'object') ? type : { type, payload }, '*');
+  ? (type, payload?) => broadcastChannel.postMessage((typeof type === 'object') ? type : { type, payload })
+  : (type, payload?) => window.parent.postMessage((typeof type === 'object') ? type : { type, payload }, '*');
 
-export function fromTopic(type) {
+export function fromTopic(type: string) {
   return message$.pipe(
-    filter((e) => e.data?.type === type),
+    filter((e: any) => e.data?.type === type),    // TODO: e type
```

Comment:
Let's stop adding TODOs all over the place and start typing. This is what this ticket is about.